src/backend/marsha/core/tests/test_api_video.py

```diff
@@ -1507,6 +1604,138 @@ def test_api_video_update_detail_token_user_title(self):
+        self.assertTrue(video.is_scheduled)
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+
+        # try to set a date in the past
+        data = {"title": "title required", "starting_at": "2000-01-01 00:00:15"}
+        response = self.client.put(
+            "/api/videos/{!s}/".format(video.id),
+            json.dumps(data),
+            HTTP_AUTHORIZATION="Bearer {!s}".format(jwt_token),
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 400)
+        # data didn't change and an error is thrown
+        self.assertEqual(
+            json.loads(response.content),
+            {
+                "starting_at": [
+                    "2000-01-01 00:00:15+00:00 is not a valid date, date should be planned after!"
+                ]
+            },
+        )
+        video.refresh_from_db()
+        self.assertEqual(video.starting_at, starting_at)
+        self.assertTrue(video.is_scheduled)
+
+    def test_api_video_update_detail_token_scheduled_date_to_none(self):
+        """
+        Update starting_at to None.
+
+        Users can update the starting_date of their video through the API to None to cancel the
+        scheduled mode.
+        """
+        starting_at = timezone.now() + timedelta(minutes=10)
+        video = factories.VideoFactory(title="my title", starting_at=starting_at)
+        self.assertTrue(video.is_scheduled)
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+
+        data = {"title": "title required", "starting_at": None}
+        response = self.client.put(
+            "/api/videos/{!s}/".format(video.id),
+            json.dumps(data),
+            HTTP_AUTHORIZATION="Bearer {!s}".format(jwt_token),
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+        video.refresh_from_db()
+        self.assertEqual(video.starting_at, None)
+        self.assertFalse(video.is_scheduled)
+
+    def test_api_video_update_detail_token_scheduled_with_previous_starting_at_already_past(
+        self,
+    ):
+        """starting_at is in the past, it can't be updated anymore."""
+        video = factories.VideoFactory(
+            starting_at=datetime(2100, 2, 4, 0, 0, tzinfo=pytz.utc)
+        )
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+        # now is set to 2111 which is after video.starting_at
```

Comment:
here it's `2111` and line after `2101`. 

Should not be easier to work with `datetime + timedelta` ?